File: Pawsome-Windows/Pawsome.Core/PawsomeConfig.cs
namespace Pawsome.Core;

/// <summary>
/// Central configuration for the Pawsome Windows client.
/// New Firebase / Google Cloud credentials will be added later.
/// </summary>
public static class PawsomeConfig
{
    // Firebase project credentials intentionally cleared during backend reset.
    public const string FirebaseProjectId = "";
    public const string FirebaseApiKey = "";
    public const string FirebaseAuthDomain = "";

    // Google OAuth client ID is supplied through the environment at runtime.
    public static string? GoogleDesktopClientId =>
        Environment.GetEnvironmentVariable("PAWSOME_GOOGLE_CLIENT_ID");

    public const string GitHubAssetsRepo = "baldbuffalo/Pawsome-assets";

    public static string FirestoreBaseUrl =>
        $"https://firestore.googleapis.com/v1/projects/{FirebaseProjectId}/databases/(default)/documents";

    public static string IdentityToolkitBaseUrl =>
        "https://identitytoolkit.googleapis.com/v1/accounts";

    public static string SecureTokenUrl =>
        $"https://securetoken.googleapis.com/v1/token?key={FirebaseApiKey}";
}

File: Pawsome-Android/app/src/main/java/com/pawsome/app/PawsomeConfig.kt
package com.example.pawsome

object PawsomeConfig {
    const val projectId = "pawsome--signin-ios"
    const val apiKey = "AIzaSyBl9nk70Vocx5nTtG4ctF5HeKazSrSSfHA"
    const val githubRepo = "baldbuffalo/Pawsome-assets"

    val googleServerClientId: String get() = BuildConfig.GOOGLE_SERVER_CLIENT_ID
    val twitterConsumerKey: String get() = BuildConfig.TWITTER_CONSUMER_KEY
    val twitterConsumerSecret: String get() = BuildConfig.TWITTER_CONSUMER_SECRET
    val githubToken: String get() = BuildConfig.GITHUB_TOKEN

    val firestoreBase: String
        get() = "https://firestore.googleapis.com/v1/projects/$projectId/databases/(default)/documents"
    const val identityBase = "https://identitytoolkit.googleapis.com/v1/accounts"
    val secureTokenUrl: String get() = "https://securetoken.googleapis.com/v1/token?key=$apiKey"
}
